src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/SqlConfigurableRetryLogicManager.cs

```diff
@@ -0,0 +1,97 @@
+
+        private static readonly Lazy<SqlConfigurableRetryLogicManager> s_instance =
+            new Lazy<SqlConfigurableRetryLogicManager>(() => new SqlConfigurableRetryLogicManager());
+
+        private SqlRetryLogicBaseProvider s_connectionProvoder = null;
+        private SqlRetryLogicBaseProvider s_commandProvider = null;
+
+        /// <summary>
+        /// The default none retry provider will apply if a parameter passes by null.
+        /// </summary>
+        private void AssignProviders(SqlRetryLogicBaseProvider cnnProvider = null, SqlRetryLogicBaseProvider cmdProvider = null)
+        {
+            s_connectionProvoder = cnnProvider ?? SqlConfigurableRetryFactory.CreateNoneRetryProvider();
+            s_commandProvider = cmdProvider ?? SqlConfigurableRetryFactory.CreateNoneRetryProvider();
+        }
+
+        /// <summary>
+        /// Default Retry provider for SqlConnections
+        /// </summary>
+        internal static SqlRetryLogicBaseProvider ConnectionProvider
+        {
+            get
+            {
+                try
+                {
+                    SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Requested the {1} value."
+                                                            , s_typeName, nameof(ConnectionProvider));
+                    return s_instance.Value.s_connectionProvoder;
+                }
+                catch (Exception e)
+                {
+                    StackTrace stackTrace = new StackTrace();
+                    SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> An exception occurred in access to the instance of the class, and the default none-retriable provider will apply: {2}"
+                                                            , s_typeName, nameof(ConnectionProvider), e);
+                    SqlClientEventSource.Log.TryAdvancedTraceEvent("<sc.{0}.{1}|ADV|INFO> An exception occurred in access to the instance of the class, and the default none-retriable provider will apply: {2}"
+                                                            , s_typeName, nameof(ConnectionProvider), stackTrace);
+                    return SqlConfigurableRetryFactory.CreateNoneRetryProvider();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Default Retry provider for SqlCommands
+        /// </summary>
+        internal static SqlRetryLogicBaseProvider CommandProvider
+        {
+            get
+            {
+                try
+                {
+                    SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Requested the {1} value."
+                                                            , s_typeName, nameof(CommandProvider));
+                    return s_instance.Value.s_commandProvider;
+                }
+                catch (Exception e)
+                {
+                    StackTrace stackTrace = new StackTrace();
```

Comment:
This is an expensive operation I think for every exception we would retry on, plus you only need this for advanced trace, not for regular trace.

Can we do something directly in TryAdvancedTraceEvent or check if advanced tracing is enabled before loading this here?